Gateway/Request/CustomerIpDataBuilder.php

```diff
@@ -52,7 +53,12 @@ public function build(array $buildSubject)
     {
         /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
-        $shopperIp = $paymentDataObject->getPayment()->getOrder()->getXForwardedFor();
+        $order = $paymentDataObject->getPayment()->getOrder();
+        $shopperIp = $order->getXForwardedFor();
+
+        if (empty($shopperIp)) {
+            $shopperIp = $order->getRemoteIp();
```

Comment:
```suggestion
            $shopperIp = $order-> getXForwardedFor();
```

`getXForwardedFor` uses the `$_SERVER['HTTP_X_FORWARDED_FOR']` server variable, which can be overwritten by the client and should not be trusted.